packages/components/src/ConfirmationModal/tests/interaction-ConfirmationModal.test.tsx

```diff
@@ -54,6 +54,35 @@ test("simple ConfirmationModal should cancel", () => {
   expect(requestCloseHandler).toHaveBeenCalled();
 });
 
+test("confirm action should not happen during a render cycle", () => {
+  jest.spyOn(console, "error");
+
+  function StatefulWrapper() {
+    const [foo, setFoo] = React.useState(true);
+
+    return (
+      <>
+        {foo ? "FOO TRUE" : "FOO FALSE"}
+        <ConfirmationModal
+          title="Should we?"
+          message="Do something…"
+          open={true}
+          confirmLabel="We Shall"
+          onConfirm={() => setFoo(false)}
+          onCancel={() => null}
+          onRequestClose={() => null}
+        />
+      </>
+    );
+  }
+
+  const { getByText } = render(<StatefulWrapper />);
+
+  fireEvent.click(getByText("We Shall"));
+  expect(getByText("FOO FALSE")).toBeInTheDocument();
+  expect(console.error).not.toHaveBeenCalled();
+});
```

Comment:
Nice to have a test to prevent future regressions!

Pontificating just to understand your perspective here... As written, the test strikes me as a bit blunt... effectively we're checking that NO error is logged to the console. That said, it's equally or more brittle to try to detect that a particular error string is not logged... in addition to the overall suspect nature of tests expecting something to NOT happen... Likely these issues have already occurred to you, so mostly curious to hear your reasoning behind this current solution. 

I do wonder if the test name could be more explicit to indicate what is being expected... 
```
test("confirm action should not happen during a render cycle and cause a React warning"
```

Or perhaps a comment above the last line to explain that a React error shouldn't happen? 

All just suggestions... let me know what you think.

Reply:
Hey @JeffMarvin,

My perspective is that this test would be characterized as a 'regression test'. i.e. a test that exists to catch a very specific bug. As such it's a little bit odd. Had we not introduced this bug in the first place, we certainly wouldn't have written the test. You could imagine infinite _hypothetical_ regressions that we could write tests for for things that do NOT happen, as you say. Instead we just write tests for the regressions we find. The 'write a failing test that exhibits the bug, and then fix it so the test passes'.

Honestly, I haven't written so many regression tests that I have a ready pattern for it. In this case, I also contemplated
- making a new file `regressions-ConfirmationModal.test.tsx`
- adding a describe block `describe('regressions', ...)`

In this case, the problem is not the React warning. The React warning is merely the manifestation of the problem. The problem is calling `setState` inside the render cycle. But adding a note about the warning certainly makes it a bit clearer.

And of course, now any future devs who are curious can just look up this PR and get this discussion, too. 😅 

I've updated the PR with your suggestion for the `it` message in b65222ab4cd445e17b0b0e9c4e27d2d42607c0e0

Happy to discuss this more! Maybe there's a better pattern we can identify for this kind of regression test.

Reply:
That all makes sense -- appreciate the additional context. I think trying to put a patten in for regression tests is a problem we could solve _after_ we achieve the institutional behavior in for adding regression tests when we find them.